src/Microsoft.ML/CSharpApi.cs

```diff
@@ -1328,40 +1413,185 @@ public sealed class Output
     namespace Data
     {
 
-        /// <summary>
-        /// Import a dataset from a text file
-        /// </summary>
-        public sealed partial class TextLoader
+        public sealed partial class TextLoaderArguments
         {
+            /// <summary>
+            /// Use separate parsing threads?
+            /// </summary>
+            public bool UseThreads { get; set; } = true;
 
+            /// <summary>
+            /// File containing a header with feature names. If specified, header defined in the data file (header+) is ignored.
+            /// </summary>
+            public string HeaderFile { get; set; }
 
             /// <summary>
-            /// Location of the input file
+            /// Maximum number of rows to produce
             /// </summary>
-            public Var<Microsoft.ML.Runtime.IFileHandle> InputFile { get; set; } = new Var<Microsoft.ML.Runtime.IFileHandle>();
+            public long? MaxRows { get; set; }
 
             /// <summary>
-            /// Custom schema to use for parsing
+            /// Whether the input may include quoted values, which can contain separator characters, colons, and distinguish empty values from missing values. When true, consecutive separators denote a missing value and an empty value is denoted by "". When false, consecutive separators denote an empty value.
             /// </summary>
-            public string CustomSchema { get; set; }
+            public bool AllowQuoting { get; set; } = true;
+
+            /// <summary>
+            /// Whether the input may include sparse representations
+            /// </summary>
+            public bool AllowSparse { get; set; } = true;
 
+            /// <summary>
+            /// Number of source columns in the text data. Default is that sparse rows contain their size information.
+            /// </summary>
+            public int? InputSize { get; set; }
 
-            public sealed class Output
-            {
-                /// <summary>
-                /// The resulting data view
-                /// </summary>
-                public Var<Microsoft.ML.Runtime.Data.IDataView> Data { get; set; } = new Var<Microsoft.ML.Runtime.Data.IDataView>();
+            /// <summary>
+            /// Source column separator.
+            /// </summary>
+            public char[] Separator { get; set; } = { '	' };
```

Comment:
>	 [](start = 55, length = 1)

Oh that's *nasty*... :) if we had declared `\n` in some other argument with type `char` we'd have parse errors. :D

Can we make the code generator robust to this? The offending line in `CSharpApiGenerator.cs` is 267. It should escape things that need escaping. (Tabs don't "need" escaping, but this looks silly.)

The idea would be, a simple utility method there that takes a character and, if necessary, escapes it. We can probably get away with anything likely to ever occur in any sort of default (`\t`, `\n`, `\r` come to mind), and just returning the char otherwise.

Previously it looks like only one other thing exposed `char` specifically, but there it was `w` so this was not evident.